### PR TITLE
[test] Add test for gperftools 

### PR DESCRIPTION
**Pull request by jgphpc**
https://github.com/eth-cscs/production/tree/master/easybuild/easyconfigs/g/gperftools

### CODE REVIEW

**Comment by jgphpc** [comment]
install was missing on dom
retest this please

**Comment by teojgo** [comment]
@jenkins-cscs retry all

**Comment by jgphpc** [comment]
There is an issue on dom with the installpath.

File: cscs-checks/tools/profiling_and_debugging/gperftools.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+class GperftoolsCheck(rfm.RegressionTest):
+    def __init__(self, lang):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu',
+                              'daint:mc', 'dom:mc']
+
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+
+        # NOTE: Reduce time limit because for PrgEnv-pgi even if the output
```
Comment (teojgo) [inline cscs-checks/tools/profiling_and_debugging/gperftools.py]: The time limit hack is not needed for this test. It runs fine without it.

**Comment by jgphpc** [comment]
https://github.com/eth-cscs/production/pull/976

File: cscs-checks/tools/profiling_and_debugging/gperftools.py (hunk 1 above)
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/gperftools.py]: https://github.com/eth-cscs/reframe/pull/670/commits/70bb7ad6c774c66f945b0ef128f8a1562a9c7339

File: cscs-checks/tools/profiling_and_debugging/gperftools.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(*([lang] for lang in ['C', 'Cpp', 'F90']))
+class GperftoolsCheck(rfm.RegressionTest):
+    def __init__(self, lang):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc']
+
```
Comment (teojgo) [inline cscs-checks/tools/profiling_and_debugging/gperftools.py]: delete this line

File: cscs-checks/tools/profiling_and_debugging/gperftools.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+
```
Comment (teojgo) [inline cscs-checks/tools/profiling_and_debugging/gperftools.py]: delete this line

File: cscs-checks/tools/profiling_and_debugging/gperftools.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            % self.rpt_file_txt
+        ]
+
+    def setup(self, environ, partition, **job_opts):
+        super().setup(environ, partition, **job_opts)
+        flags = self.prgenv_flags[self.current_environ.name]
+        self.build_system.cflags = flags
+        self.build_system.cxxflags = flags
+        self.build_system.fflags = flags
+        self.ldflags = [' -dynamic `pkg-config --libs libprofiler`']
```
Comment (teojgo) [inline cscs-checks/tools/profiling_and_debugging/gperftools.py]: no need to define `self.ldflags`, add the list directly in the following line.

**Pull request merged**